src/test/java/reporter/CustomReporter.java

```diff
@@ -0,0 +1,117 @@
+package reporter;
+
+import TestRailIntegration.TestRailIntegration;
+import com.gurock.testrail.APIClient;
+import lombok.SneakyThrows;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.testng.*;
+import org.testng.xml.XmlSuite;
+import slackIntegration.EyesSlack;
+
+import java.util.*;
+
+public class CustomReporter extends TestRailIntegration implements IReporter {
+
+    private static Map<Integer, String> idAndComment = new HashMap<>();
+
+    static {
+        idAndComment.put(1, "Current status is : Passed");
+        idAndComment.put(5, "Current status is : Failed");
+    }
+
+    private final Map<Integer, Long> statusIdAndCaseId = new HashMap<>();
+    private final Map<Integer, List<String>> passedTC = new HashMap<>();
+    private final Map<Integer, List<String>> failedTC = new HashMap<>();
+    private final Map<Integer, List<String>> skippedTC = new HashMap<>();
+
+
+    @SneakyThrows
+    @Override
+    public void generateReport(List<XmlSuite> xmlSuites, List<ISuite> suites,
+                               String outputDirectory) {
+
+        //Iterating over each suite included in the test
+        for (ISuite suite : suites) {
+
+            //Following code gets the suite name
+            String suiteName = suite.getName();
+
+            //Getting the results for the said suite
+            Map<String, ISuiteResult> suiteResults = suite.getResults();
+            for (ISuiteResult sr : suiteResults.values()) {
+                ITestContext tc = sr.getTestContext();
+                List<String> allTestCasesInSuite =
+                        Arrays.stream(tc.getAllTestMethods()).map(ITestNGMethod::getMethodName).toList();
+                List<String> passedTests =
+                        tc.getPassedTests().getAllResults().stream().map(ITestResult::getName).toList();
+                passedTC.put(1, passedTests);
+                List<String> failedTests =
+                        tc.getFailedTests().getAllResults().stream().map(ITestResult::getName).toList();
+                failedTC.put(5, failedTests);
+                List<String> skippedTests =
+                        tc.getSkippedTests().getAllResults().stream().map(ITestResult::getName).toList();
+                skippedTC.put(6, skippedTests);
+                System.out.println("Passed tests for suite '" + suiteName +
```

Comment:
use logging to Consol instead of System.out.println

Reply:
Done